AsyncDisplayKit/TextKit/ASTextKitRenderer.mm

```diff
@@ -141,7 +138,8 @@ - (void)setConstrainedSize:(CGSize)constrainedSize
 - (void)_calculateSize
 {
   [self truncater];
-  if ([_attributes.pointSizeScaleFactors count] > 0) {
+  // if we have no scale factors or an unconstrained width, there is no reason to try to adjust the font size
+  if ([_attributes.pointSizeScaleFactors count] > 0 && isinf(_constrainedSize.width) == NO) {
```

Comment:
For both this condition and the one below, we might as well put the scaleFactors count check at the end so we can short circuit evaluation faster - this is particularly true for the second one because it will be more common for _sizeIsCalculated == YES there, and rendering on main is common (this will be enabled by default for visible range of tables / collections as they are presented by tab switch, modal, etc)